File: io-hotmoka-network/src/main/java/io/hotmoka/network/responses/JarStoreTransactionFailedResponseModel.java
package io.hotmoka.network.responses;

import java.math.BigInteger;

import io.hotmoka.beans.responses.JarStoreTransactionFailedResponse;
import io.hotmoka.network.updates.UpdateModel;

public class JarStoreTransactionFailedResponseModel extends JarStoreTransactionResponseModel {

    /**
     * The amount of gas consumed by the transaction as penalty for the failure.
     */
    public String gasConsumedForPenalty;

    /**
     * The fully-qualified class name of the cause exception.
     */
    public String classNameOfCause;

    /**
     * The message of the cause exception.
     */
    public String messageOfCause;

    public JarStoreTransactionFailedResponseModel(JarStoreTransactionFailedResponse response) {
        super(response);

        this.gasConsumedForPenalty = response.gasConsumedForPenalty().toString();
        this.classNameOfCause = response.classNameOfCause;
        this.messageOfCause = response.messageOfCause;
    }

    public JarStoreTransactionFailedResponseModel() {}

    public JarStoreTransactionFailedResponse toBean() {
        return new JarStoreTransactionFailedResponse(
        	classNameOfCause,
        	messageOfCause,
        	updates.stream().map(UpdateModel::toBean),
        	new BigInteger(gasConsumedForCPU),
        	new BigInteger(gasConsumedForRAM),
        	new BigInteger(gasConsumedForStorage),
        	new BigInteger(gasConsumedForPenalty)
        );
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-js/src/models/responses/JarStoreTransactionFailedResponseModel.ts
import {JarStoreTransactionResponseModel} from "./JarStoreTransactionResponseModel";
import {UpdateModel} from "../updates/UpdateModel";

export class JarStoreTransactionFailedResponseModel extends JarStoreTransactionResponseModel {
    /**
     * The amount of gas consumed by the transaction as penalty for the failure.
     */
    gasConsumedForPenalty: string

    /**
     * The fully-qualified class name of the cause exception.
     */
    classNameOfCause: string

    /**
     * The message of the cause exception.
     */
    messageOfCause: string

    constructor(
        updates: Array<UpdateModel>,
        gasConsumedForCPU: string,
        gasConsumedForRAM: string,
        gasConsumedForStorage: string,
        gasConsumedForPenalty: string,
        classNameOfCause: string,
        messageOfCause: string
    ) {
        super(updates, gasConsumedForCPU, gasConsumedForRAM, gasConsumedForStorage)
        this.gasConsumedForPenalty = gasConsumedForPenalty
        this.classNameOfCause = classNameOfCause
        this.messageOfCause = messageOfCause
    }
}